### ISSUE TITLE
Workspaces not restored after VSCode update in windows displaying "Release Notes" tab

### ISSUE THREAD

**Issue by stefcameron** [Microsoft/vscode#56575]
I searched open issues, didn't find anything like this.

I'm not sure how to check this on the Insiders build...

- VSCode Version: 1.26.0
- OS Version: macOS 10.13.6

Steps to Reproduce:

1. Quit VSCode while multiple windows (and therefore workspaces) are open. Each workspace has a single folder open within it, each a different folder.
2. This time, I rebooted my laptop. Not because I thought I needed to, but because I had other updates I needed to install. I didn't know VSCode would also be updated.
3. Once rebooted, start VSCode, which happened to install the latest 1.26.0 update (I had been on 1.25.x).

4 of the 7 windows/workspaces I had open were correctly restored, but did not show the "Release Notes: 1.26.0" tab. The other 3 did show the "Release Notes" tab, but did not restore my workspaces.

So for the windows that ended-up showing the "Release Notes" tab, VSCode did not restore their respective workspaces (correct folders open, but no previously-open files and tab layouts).

Does this issue occur when all extensions are disabled?: Too late to tell since VSCode has already been updated.

**Comment by bpasero** [comment]
@stefcameron how much time ellapsed between shutdown of VSCode and restart of the OS? Maybe Code was still busy storing UI state even though no window was visible.

**Comment by stefcameron** [comment]
@bpasero Good question! I would say about a minute, not much more. What's curious to me is that only 3 of the 7 windows displayed the special Release Notes tab after OS restart and manual VSCode relaunch. Those 3 didn't restore workspaces. The other 4 restored their workspaces, but didn't show the special Release Notes tab. Why did this split happen, and what determined those 3 only and not the others? Is there a process in showing the Release Notes tab that prevents the workspace from being restored properly?

**Comment by bpasero** [comment]
@stefcameron not sure, does this reproduce reliably even now that you updated? e.g. I would not expect the release notes tab to open each time (just once after an update).

**Comment by stefcameron** [comment]
@bpasero I hadn't upgraded at home yet (different laptop, same OS, same initial VSCode version at 1.25.x), and I had 2 workspaces/windows open. So I quit VSCode, waited about 30 seconds, checked there was no longer a black dot beneath the VSCode icon in the dock, and then relaunched VSCode. Then I got this error message:

![after-relaunch-vscode](https://user-images.githubusercontent.com/2855350/44303964-12824200-a314-11e8-98d1-b3c2aaaaf7c9.png)

First time. I had never seen that before. My stomach turned a few times because one of my workspaces was quite customized (a 3-column with the center column horizontally split, and a bunch of files open everywhere, with the console in vertical mode). The only thing I could do from this dialog was choose "Close" because "Reopen" wasn't doing anything. That caused VSCode to close/quit. There was no longer a black dot beneath its icon in the dock after that.

So I relaunched again. This time, VSCode came up. My large workspace was fully restored (😌), but the Release Notes tab didn't show-up in it. My other workspace was completely blank, like I had never opened that folder. I checked the VSCode version, and I was still at 1.25.x (the latest patch before 1.26.0). So I quit again, hoping to trigger the upgrade in the same way as originally.

When I relaunched again, my large workspace came back up, no Release tab. My other workspace, which was now "empty" (reset from the error that occurred) had the Release tab displayed in it. I'm now upgraded to 1.26.0.

I have screenshots of my workspaces before, in the middle, and after, if what I wrote here isn't descriptive enough. I'm not sure you'll be able to conclude anything from this, though, but I still feel like there's an issue here, possibly a couple in light of the error I experienced this time.

**Comment by stefcameron** [comment]
@bpasero I followed your instructions on both laptops and all workspaces were restored correctly. I tried this at home yesterday, at which point it seems 1.26.1 wasn't posted yet because an upgrade didn't happen. When I tried this at work just now, 1.26.1 was posted, so the upgrade took place when I relaunched VSCode. The Release tab showed-up in 2 out of 6 workspaces, but this time, all workspaces were correctly restored. `¯\_(ツ)_/¯`
